RULES_DESCRIPTIONS.md

```diff
@@ -122,6 +123,19 @@ The garbage collector can be configured through environment variables as describ
 
 _Configuration_: N/A
 
+## cognitive-complexity
+
+_Description_: [Cognitive complexity](https://www.sonarsource.com/resources/white-papers/cognitive-complexity.html) is a measure of how hard code is to understand. Enforcing a maximum complexity per function helps to keep code readable and maintainable.
```

Comment:
Maybe one sentence on how it differs from cyclomatic complexity?

Reply:
Done.